### PR TITLE
Update to store file viewer to include new share link modals

### PR DESCRIPTION
**Pull request by cj-f**

### CODE REVIEW

File: packages/react-mui-branding/src/components/BrandingThemeProvider/index.tsx [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import { FC, useMemo } from "react";
+import merge from "lodash/merge";
+import cloneDeep from "lodash";
```
Comment (kenany) [inline packages/react-mui-branding/src/components/BrandingThemeProvider/index.tsx:3]: `cloneDeep` is never used.

File: packages/react-mui-branding/src/components/BrandingThemeProvider/index.tsx [hunk 2]
```diff
@@ -34,8 +36,9 @@ const BrandingThemeProvider: FC<BrandingThemeProviderProps> = ({
       const themeOptions_ = getBrandingThemeOptions(
         configProp ?? (config || {})
       );
-      return createTheme({ ...themeOptions_, ...themeOptions });
-    } catch {
+      return createTheme(merge({ ...themeOptions_ }, { ...themeOptions }));
+    } catch (e) {
+      console.error(e);
```
Comment (kenany) [inline packages/react-mui-branding/src/components/BrandingThemeProvider/index.tsx:41]: Stray `console.log()`?

File: packages/react-mui-store-file-viewer/src/components/CreateNewShareButton/CreateSharesButton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+      .create({
+        links: (body?.authenticators ?? [])?.map((auth) => {
+          return {
+            ...body,
+            targets: [object.id],
+            authenticators: [auth],
+          };
+        }),
+      })
+      .then(async (response) => {
```
Comment (kenany) [inline packages/react-mui-store-file-viewer/src/components/CreateNewShareButton/CreateSharesButton/index.tsx:134]:
```suggestion
      .then((response) => {
```

File: packages/react-mui-store-file-viewer/src/components/CreateNewShareButton/CreateSharesButton/index.tsx [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+        const links = response?.result?.share_link_objects ?? [];
+        if (links.length) {
+          setOpen(false);
+          setSending(links);
+        }
+
+        return response;
+      })
+      .catch((err) => {
+        throw err;
```
Comment (kenany) [inline packages/react-mui-store-file-viewer/src/components/CreateNewShareButton/CreateSharesButton/index.tsx:147]: Don't think we need this `catch` if all it's gonna do is re-throw.

File: packages/react-mui-store-file-viewer/src/components/CreateNewShareButton/SendShareViaEmailButton/SendShareViaEmailModal.tsx [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    );
+  }, [shares]);
+
+  const objAuthType = Object.values(obj)?.[0]?.type;
+
+  const handleClose = () => {
+    onClose();
+  };
+
+  const handleSendShare = async (data: Record<string, ShareSendObj>) => {
```
Comment (kenany) [inline packages/react-mui-store-file-viewer/src/components/CreateNewShareButton/SendShareViaEmailButton/SendShareViaEmailModal.tsx:106]:
```suggestion
  const handleSendShare = (data: Record<string, ShareSendObj>) => {
```

**Review by kenany — APPROVED** [review]

File: packages/react-mui-branding/src/components/BrandingThemeProvider/index.tsx (hunk 2 above)
Comment (cj-f) [reply packages/react-mui-branding/src/components/BrandingThemeProvider/index.tsx:41]: No this was an intentional error, we don't want this throwing an uncaught exception, but we at least want to console error log so it is somewhat visible in development

File: packages/react-mui-store-file-viewer/src/components/CreateNewShareButton/SendShareViaEmailButton/SendShareViaEmailModal.tsx (hunk 5 above)
Comment (cj-f) [reply packages/react-mui-store-file-viewer/src/components/CreateNewShareButton/SendShareViaEmailButton/SendShareViaEmailModal.tsx:106]: This async was wanted, I want this function to always return a promise, but there is an early exit condition that just returns, but I don't want that changing the function from returning a promise

**Pull request merged**